### PR TITLE
refactor: update app.py

### PR DESCRIPTION
**Pull request by drduhe**
**Issue #, if available:** n/a

### Notes
This PR introduces significant improvements to the ModelRunner class by enhancing its internal logic, refactoring its structure for better maintainability, and implementing comprehensive unit tests to ensure robustness. The changes aim to improve error handling, code readability, and test coverage.

#### Key Changes:

1. Refactor of ModelRunner Class:

	•	Improved method separation and encapsulation of logic to make the class easier to read and maintain.
	•	Enhanced exception handling for common scenarios like retryable job failures (RetryableJobException) and throttling delays (SelfThrottledRegionException).
	•	Added more detailed logging to capture important events during image and region request processing, facilitating easier debugging.
	•	Improved flow control in _process_image_requests and _process_region_requests to better handle errors and edge cases.

2. Implementation of Unit Tests:

	•	Test Coverage: Added unit tests for key scenarios including:
	•	Successful image and region request processing.
	•	Handling invalid image requests with InvalidImageRequestException.
	•	Retryable errors (RetryableJobException) leading to reset requests.
	•	Throttled region requests leading to a delay using SelfThrottledRegionException.
	•	General exception handling and logging in both image and region request processing.
	•	Mocking: Used unittest.mock to simulate interactions with external services (SQS, GDAL, etc.) without relying on actual resources.
	•	Edge Case Handling: Unit tests cover a variety of failure cases, ensuring the class can handle invalid data and unexpected conditions.

3. `update_custer.sh`
	•	Implemented ECS Service Management Script: Added a bash script to streamline the management of ECS services by setting the desired task count to 0 and disabling autoscaling for services matching a specified pattern within a cluster. The script supports configurable parameters for desired task count, cluster name, service pattern, and AWS region, ensuring flexibility and ease of use. Example usage and validation steps are included to improve debugging and operational efficiency.

### Checklist

Before you submit a pull request, please make sure you have the following:
- [x] Code changes are compact and well-structured to facilitate easy review
- [x] Changes are documented in the README.md and other relevant documentation pages
- [x] PR title and description accurately reflect the changes and are detailed enough for historical tracking
- [x] PR contains tests that cover all new code and the code has been manual tested
- [x] All new dependencies are declared (if any), and no unnecessary libraries are added
- [x] Performance impacts (if any) of the changes are evaluated and documented
- [x] Security implications of the changes (if any) are reviewed and addressed
- [x] I have read the [Contributing Guidelines](../CONTRIBUTING.md) and agree to follow the [Code of Conduct](../CODE_OF_CONDUCT.md)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/aws/osml/model_runner/model_runner.py [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    def monitor_work_queues(self) -> None:
+        """
+        Continuously monitors the SQS queues for RegionRequest and ImageRequest.
+        :return: None
+        """
+        set_gdal_default_configuration()
+
+        while self.running:
+            try:
+                self._process_region_requests()
```
[the remark this answers is not in the document]
Comment (drduhe) [reply src/aws/osml/model_runner/model_runner.py:109]: Updated logic

File: src/aws/osml/model_runner/model_runner.py [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+        if region_request_item is None:
+            region_request_item = RegionRequestItem.from_region_request(region_request)
+            self.region_request_table.start_region_request(region_request_item)
+            logger.debug(
+                f"Added region request: image id {region_request_item.image_id}, region id {region_request_item.region_id}"
+            )
+        return region_request_item
+
+    @staticmethod
+    def _get_image_path(region_request: RegionRequest) -> str:
```
[the remark this answers is not in the document]
Comment (drduhe) [reply src/aws/osml/model_runner/model_runner.py:227]: Migrating

**Review by edparris — APPROVED** [review]

**Review by RanbirAulakh — APPROVED** [review]
